src/pyodide/internal/snapshot.ts

```diff
@@ -322,11 +334,20 @@ function memorySnapshotDoImports(Module: Module): string[] {
  * linear memory into MEMORY.
  */
 function makeLinearMemorySnapshot(Module: Module): Uint8Array {
-  const dsoJSON = recordDsoHandles(Module);
+  let json: SnapshotJson = recordDsoHandles(Module);
+  if (Module.API.version !== '0.26.0a2') {
+    const hiwire = Module.API.serializeHiwireState(Module);
+    const tmp: NewSnapshotJson = {
+      version: 1,
+      dsos: json,
+      hiwire,
+    };
+    json = tmp;
```

Comment:
Nit

```suggestion
    json = {
      version: 1,
      dsos: json,
      hiwire,
    } as NewSnapshotJson;
```

Reply:
If you do:
```ts
const a: A = expr;
```
then typescript will check that expr is a subtype of A. If you do:
```ts
const a = expr as A;
```
then typescript will not check the type of `expr`. Thus the former is to be preferred whenever it works, and using `as` is a fallback for when we can't convince typescript to type check stuff properly.